app/src/components/settings/settings_view/index.tsx

```diff
@@ -0,0 +1,254 @@
+  const history = useHistory()
+  const context = useConnectedWeb3Context()
+
+  const [current, setCurrent] = useState(0)
+  const [url, setUrl] = useState<string>('')
+  const [onlineStatus, setOnlineStatus] = useState<boolean>(false)
+
+  const urlObject = getChainSpecificAlternativeUrls(context.networkId)
+
+  const dropdownItem = urlObject.map((item, index) => {
+    return {
+      title: item.name,
+      image: item.name === 'Infura' ? imgInfura : undefined,
+      onClick: () => {
+        setCurrent(index)
+        setUrl(item.rpcUrl)
+      },
+    }
+  })
+
+  dropdownItem.push({
+    title: 'Custom',
+    image: undefined,
+    onClick: () => {
+      setCurrent(dropdownItem.length - 1)
+    },
+  })
+
+  const filterItems = dropdownItem.map(item => {
+    // console.log(index)
+    return {
+      content: (
+        <CustomDropdownItem onClick={item.onClick}>
+          {item.image && <img alt="node" src={item.image} />}
+          {item.title}
+        </CustomDropdownItem>
+      ),
+      secondaryText: '',
+      onClick: item.onClick,
+    }
+  })
+  const checkRpcStatus = async () => {
+    try {
+      await axios.post(url, {
+        id: +new Date(),
+        jsonrpc: '2.0',
+        method: 'eth_getBlockByNumber',
+        params: ['latest', true],
+      })
+
+      setOnlineStatus(true)
+      return true
+    } catch (e) {
+      setOnlineStatus(false)
+
+      return false
+    }
+  }
+  useEffect(() => {
+    if (url.length === 0 && current !== dropdownItem.length - 1) {
+      setUrl(urlObject[current].rpcUrl)
+    }
+    if (!isValidHttpUrl(url)) {
+      return
+    }
+
+    checkRpcStatus()
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [url])
+
+  return (
+    <ListCard style={{ minHeight: 'initial' }}>
+      <TopContent>
+        <Text>Settings</Text>
+      </TopContent>
+
+      <MainContent>
+        <Row>
+          <Column>
+            <Text>RPC Endpoint</Text>
+            <StatusSection>
+              <StatusBage status={onlineStatus} />
+              <TextLighter>Status: {onlineStatus ? 'OK' : 'Unavailable'}</TextLighter>
+            </StatusSection>
+          </Column>
+          <FiltersControls>
+            <NodeDropdown currentItem={current} dirty dropdownPosition={DropdownPosition.center} items={filterItems} />
+          </FiltersControls>
+        </Row>
+        {current === dropdownItem.length - 1 && (
+          <Input onChange={event => setUrl(event.target.value)} placeholder="Paste your RPC URL"></Input>
+        )}
+      </MainContent>
+
+      <BottomContent>
+        <ButtonRound
+          onClick={() => {
+            history.push('/')
+          }}
+        >
+          Back
+        </ButtonRound>
+        <ButtonRow>
+          <ButtonRound
+            onClick={() => {
+              setCurrent(0)
+              setUrl(urlObject[0].rpcUrl)
+            }}
+          >
+            Set to Default
+          </ButtonRound>
```

Comment:
As a user, when selecting 'Set to default', I expect the same effect as selecting an endpoint and clicking save, i.e. going to a new page and reloading

Reply:
@pimato yeah we should probably agree on this since geronimo gave feedback to not change page but to stay on the current page....